File: farewell.dart
import 'dart:io';
import 'dart:async';
import 'dart:convert';

void main(List<String> arguments) async {
  var key = 'i hope in the next ten years there would be no other farewell letter brilliant than this one';
  var data = await (new File(arguments[0]).readAsString(encoding: ASCII));
  var encrypted = data.replaceAll(new RegExp(r'\s'), '').split(',');
  var decrypted = new List<int>();
  for (int i = 0; i < encrypted.length; ++i) {
    decrypted.add(int.parse(encrypted[i]) ^ key.codeUnitAt(i % key.length));
  }
  stdout.write(UTF8.decode(decrypted));
}

File: farewell.swift
import Foundation

let key = "i hope in the next ten years there would be no other farewell letter brilliant than this one".characters.map { String($0) }
let key_len = key.count

let data = try String(contentsOfFile: CommandLine.arguments[1])
let encrypted = data.trimmingCharacters(in: .whitespacesAndNewlines).components(separatedBy: ",")

var decrypted = [UInt8]()

for (index, element) in encrypted.enumerated() {
    let a = UInt32(element)
    let b = key[index % key_len].unicodeScalars.filter{$0.isASCII}.first?.value
    let c = a! ^ b!
    decrypted.append(UInt8(c))
}

print(String(bytes: decrypted, encoding: String.Encoding.utf8)!, terminator: "")
